x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlClearCursorRequest.java

```diff
@@ -43,9 +48,11 @@
         PARSER.declareString(optionalConstructorArg(), MODE);
         PARSER.declareString(optionalConstructorArg(), CLIENT_ID);
         PARSER.declareString(optionalConstructorArg(), VERSION);
+        PARSER.declareBoolean(SqlClearCursorRequest::binaryCommunication, BINARY_FORMAT);
```

Comment:
Declaring this field mandatory (instead of optional), breaks backwards compatibility for clients that do not set this parameter.

Reply:
Not sure I got it, maybe I'm misinterpreting some logic here. 
Isn't it optional already? (see also test https://github.com/elastic/elasticsearch/pull/84230/files#diff-7d14bc9a0a5fb5167a3d7eea6b457ea14324136f334a33df4417ec5f78fefb84R89-R102 )

Reply:
I read the above as having `constructorArg()` in a similar style to the previous declaration.